File: cli/src/preflights/preflight-list.ts
import chalk from "chalk";
import { loadRegistry } from "../utils/registry.js";

export interface PreflightResult {
  success: boolean;
  error?: string;
  details?: string[];
}

/**
 * Pre-flight checks for the list command
 * @returns PreflightResult with validation status and messages
 */
export function preflightList(): PreflightResult {
  try {
    // Check if registry can be loaded
    loadRegistry();
    return {
      success: true,
    };
  } catch (error) {
    return {
      success: false,
      error: chalk.red("✗ Failed to load component registry"),
      details: [
        chalk.gray(
          `  ${error instanceof Error ? error.message : String(error)}`
        ),
      ],
    };
  }
}

File: dart-cli/lib/preflights/preflight_list.dart
import 'package:fluttercn/utils/registry.dart';

/// Preflight result with success status
class PreflightResult {
  final bool success;
  final String? error;
  final List<String>? details;

  PreflightResult({
    required this.success,
    this.error,
    this.details,
  });
}

PreflightResult preflightList() {
  try {
    loadRegistry();
    return PreflightResult(success: true);
  } catch (e) {
    return PreflightResult(
      success: false,
      error: '✗ Failed to load component registry',
      details: ['  $e'],
    );
  }
}
